library/src/main/java/org/codice/security/sign/SimpleSign.java

```diff
@@ -80,9 +80,28 @@ public SimpleSign() throws IOException {
    */
   public Map<String, String> signUriString(
       String samlType, String samlRequestOrResponse, String relayState) throws SignatureException {
+    return signUriString(samlType, samlRequestOrResponse, relayState, null);
+  }
+  /**
+   * Signs uri value. According to the SAML Spec,
+   *
+   * <p>"To construct the signature, a string consisting of the concatenation of the RelayState (if
+   * present), SigAlg, and SAMLRequest (or SAMLResponse) query string parameters (each one
+   * URLencoded) is constructed in one of the following ways (ordered as below):
+   * SAMLRequest=value&RelayState=value&SigAlg=value
+   * SAMLResponse=value&RelayState=value&SigAlg=value"
+   *
+   * @param samlType - SAMLRequest or SAMLResponse
+   * @param samlRequestOrResponse - request or response already encoded
+   * @param relayState - uri encoded relayState (optional) - null is no relay state exists
+   * @param sigAlgo - The Signature algorithm to use. If null, the cert will be used to determine.
+   */
+  public Map<String, String> signUriString(
+      String samlType, String samlRequestOrResponse, String relayState, String sigAlgo)
+      throws SignatureException {
     try {
       X509Certificate[] certificates = getSignatureCertificates();
-      String sigAlgo = getSignatureAlgorithm(certificates[0]);
+      if (sigAlgo == null) sigAlgo = getSignatureAlgorithm(certificates[0]);
```

Comment:
❗️ I don't think you can explicitly set the sigAlg and just have it work. I think the algorithm we chose is dependent on the private/public signing key pair and their algorithms.